lib/db.js

```diff
@@ -479,10 +481,21 @@ module.exports = (
       uaDeviceType: token.uaDeviceType,
       lastAccessTime: token.lastAccessTime,
       createdAt: token.createdAt
-    }]
+    }
+
     let sessionTokens = []
     // get the array of session tokens associated with the given uid
-    return this.redis.getAsync(uid)
+    return getGeoData(ip)
+    .then((res) => {
+      newToken.location = res.location
+      return newToken
```

Comment:
There's already some logging in `lib/geodb.js` so let's leave it at that I reckon.

Side-note, we probably want to do this geo-lookup as infrequently as possible. There are calls to it elsewhere that we should be looking to get rid of in favour of this cached version (if it's available). Further, we should think about ways to eliminate unnecessary calls to it here if we can. For instance, if we stored a hash of the IP address we could compare against that before knowing whether we need to lookup again (using a hash so we don't store PII). Make sense?